### PR TITLE
added devicePath check for iscsi disk

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
#### What type of PR is this?
Bug Fixer 

#### What this PR does / why we need it:
It adds device path check for iscsi disk
#### Which issue(s) this PR fixes:

part of #107231 

#### Does this PR introduce a user-facing change?

```release-note
none
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/volume/iscsi/attacher.go [hunk 1]
```diff
@@ -81,6 +81,10 @@ func (attacher *iscsiAttacher) WaitForAttach(spec *volume.Spec, devicePath strin
 		klog.Warningf("failed to get iscsi mounter: %v", err)
 		return "", err
 	}
+	exists, err := mount.PathExists(devicePath)
+	if !exists && err != nil {
```
Comment (jsafrane) [inline pkg/volume/iscsi/attacher.go:85]: It does not make sense to check `devicePath`, it's always empty (as returned by `Attach()` above)

**Comment by dims** [comment]
@NikhilSharmaWe did you see the comment from reviewer above?

File: pkg/volume/iscsi/attacher.go (hunk 1 above)
Comment (NikhilSharmaWe) [reply pkg/volume/iscsi/attacher.go:85]: @jsafrane should I close this PR.
Also could you please review the PR for fc disk, https://github.com/kubernetes/kubernetes/pull/107279#

Comment (YuikoTakada) [reply pkg/volume/iscsi/attacher.go:85]: apply the same change with #107279 ?
```suggestion
	if exists && err == nil {
		return devicePath, nil
	}
	if exists == false {
		return "", fmt.Errorf("device %s does not exist", devicePath)
	} else {
		return "", err
```

Comment (jsafrane) [reply pkg/volume/iscsi/attacher.go:85]: Again, `devicePath` is always empty. It really does not make any sense to check it here.

**Comment by NikhilSharmaWe** [comment]
@jsafrane Could you please clear out whether this PR is needed or not?

**Comment by jsafrane** [comment]
No, this PR is not needed, since `devicePath` should be always empty here.
/close

**Pull request closed without merging**